src/features/references.zig

```diff
@@ -236,6 +236,18 @@ fn symbolReferences(
     switch (decl_handle.decl) {
         .ast_node => {
             try builder.collectReferences(curr_handle, 0);
+
+            const source_index = offsets.positionToIndex(curr_handle.tree.source, request.position(), encoding);
```

Comment:
The scope should be looked up where the declaration is located and not where the cursor is. The symbol/decl may be in a different file that where the request was initiated.
```suggestion
            const source_index = offsets.tokenToIndex(decl_handle.handle.tree, decl_handle.nameToken());
```

Here is an example that can cause a crash:
```zig
// A.zig
pub const Foo = struct {};
```
```zig
// B.zig
const A = @import("A.zig");
fn foo() void {
    _ = A.Foo;
    //     ^ find all references here
}
```